src/main/java/fi/fmi/avi/archiver/initializing/MessageFileMonitorInitializer.java

```diff
@@ -90,24 +103,32 @@ private static FileMetadata createFileMetadata(final Message<?> message, final F
 
     @PostConstruct
     private void initializeFilePatternFlows() {
+        final FileNameGenerator timestampAppender = msg -> msg.getHeaders().get(FileHeaders.FILENAME) + "." + clock.millis();
+
         aviationProductsHolder.getProducts().values().forEach(product -> {
             final FileReadingMessageSource sourceDirectory = new FileReadingMessageSource();
             sourceDirectory.setDirectory(product.getInputDir());
 
             final FileWritingMessageHandler archiveDirectory = new FileWritingMessageHandler(product.getArchiveDir());
             archiveDirectory.setFileExistsMode(FileExistsMode.REPLACE_IF_MODIFIED);
             archiveDirectory.setExpectReply(false);
+            archiveDirectory.setDeleteSourceFiles(true);
+            archiveDirectory.setAdviceChain(ImmutableList.of(exceptionTrapAdvice, archiveRetryAdvice));
+            archiveDirectory.setFileNameGenerator(timestampAppender);
 
             final FileWritingMessageHandler failDirectory = new FileWritingMessageHandler(product.getFailDir());
-            archiveDirectory.setFileExistsMode(FileExistsMode.REPLACE_IF_MODIFIED);
-            archiveDirectory.setExpectReply(false);
+            failDirectory.setFileExistsMode(FileExistsMode.REPLACE_IF_MODIFIED);
+            failDirectory.setExpectReply(false);
+            failDirectory.setDeleteSourceFiles(true);
+            failDirectory.setAdviceChain(ImmutableList.of(exceptionTrapAdvice, failRetryAdvice));
```

Comment:
As a small optimization, this list could be instantiated only once outside the loop.